### PR TITLE
Fix various sticky action box issues

### LINKED ISSUE #3827: Sticky box is not sticking at the right point
The sticky box should not stick until the box reaches the top of the scream on a normal scroll. Currently it's jumping on mobile and jumps on desktop after you scroll down and then back up,

### PR DESCRIPTION
**Pull request by jbalboni**
Closes #3827, #3828, #3935 

Fixes some positioning issues by grabbing the initial position of the box and using that for comparisons when scrolling. Also sets the box to full width on mobile and adds a resize handler so that it works properly when resizing between mobile and desktop.

screen shot 2016-11-29 at 12 25 30 pm
screen shot 2016-11-29 at 12 25 16 pm
screen shot 2016-11-29 at 12 25 12 pm

### CODE REVIEW

**Comment by alexose** [comment]
Looks good at first glance.  I'm going to take a bit to try it on different devices, etc.

File: src/js/common/utils/sticky-action-box.js [hunk 1]
```diff
@@ -25,22 +26,32 @@ function resetSmallActionBox(smallActionBox) {
...
     }
 
-    if (document.body.scrollTop > offset.top) {
+    if (initialPosition && document.body.scrollTop > initialPosition) {
       positionFunction(actionBox);
     } else {
       resetFunction(actionBox);
     }
   }, 100));
+  window.addEventListener('resize', _.throttle(() => {
```
Comment (alexose) [inline src/js/common/utils/sticky-action-box.js]: Never knew about `_.throttle`... cool!

**Comment by alexose** [comment]
I'm noticing that there's a minor jog that happens on mobile.  When the box pops out from `position:initial` to `position:fixed`, it goes from this:

image

to this:

image

It's a pretty minor problem, but it might be worth looking at another way to handle the width.

**Review by akainic — APPROVED** [review]
LGTM!

**Comment by jbalboni** [comment]
@alexose I see that problem when the screen width is smaller than the minimum page header width:

screen shot 2016-11-30 at 9 01 16 am

screen shot 2016-11-30 at 9 01 25 am

I think a better fix for that is to set a minimum width for the whole page or fix the page header so it collapses more, rather than changing the way the box width works. It seems pretty minor, though, as it only shows up for me on screen sizes below the iPhone 5.

I'll create a separate issue for it.

**Pull request merged**